woocommerce/wc-admin
# Add remind me later support to task list

Fixes #6637 

This replaces the Dismiss button in the task list item to an ellipsis dropdown menu, this contains the dismiss button and also a 'remind me later' option if that option is enabled.
To align the ellipsis menu properly with the header, I switched the task layout to css grid. 

### Accessibility




### Screenshots

![remind-me-later](https://user-images.githubusercontent.com/2240960/117045451-e8505880-ace5-11eb-8e98-3de224a9b0d2.gif)

### Detailed test instructions:

-  Install the task list tester plugin from here (https://github.com/louwie17/woocommerce-admin-tasklist-helper/releases/tag/v1.0.0)
- Make sure the above plugin is enabled.
- Navigate to **WooCommerce > Home** having finished the onboarding flow
- The extended task list **Things to do next** should be present.
- Click on the three dot ellipsis menu for the item `Set up the task list tester` task 
- The dropdown should include `Remind me later`, click on this
- The task should disappear and trigger the `wcadmin_extended_tasklist_remindmelater_task` event, it should also stay gone on refresh
- This should dismiss it for 24hours, but we can shorten this by updating the `woocommerce_task_list_remind_me_later_tasks` option.
- Using the [WCA test helper](https://github.com/woocommerce/woocommerce-admin-test-helper/releases/tag/v0.3.0) go to **Tools > WCA Test Helper > Options** and search for the above option
- Edit the option and change the timestamp (in milliseconds) of the `woocommerce-admin-tasklist-tester-setup` key to something more recent (you can use [epochconverter](https://www.epochconverter.com/) just append `000` at the back for milliseconds)
- After the set wait timestamp has passed and you refresh the page, the task list item should appear again.
- Try the `Remind me later` again, and click the `undo` notice, it should make the task show up again.
